src/main/scala/lms/transformation/mpi_nccl_transformation/distributedTensor2MPINCCLKernels.scala

```diff
@@ -0,0 +1,252 @@
+package lms.transformation.tensor
+
+import scala.annotation.implicitNotFound
+import scala.collection._
+import scala.collection.mutable.ListBuffer
+
+import lms.core._
+import lms.core.stub._
+import lms.collection.mutable._
+import lms.macros.SourceContext
+import lms.thirdparty.{RandomDataTypeLess, NCCLTypeLess, MPIOps, NCCLOps, SIZE_TTypeLess, CUDNNOps,CUDNNTypeLess,CLibTypeLess}
+import lms.thirdparty.array_computation.{ArrayCPUTypeLess, CUDATypeLess, CUBLASTypeLess, CudaOps, CudaLibs}
+import lms.transformation.util.{DataStructure, CudnnUtils}
+
+import Backend._
+
+
+trait DistributeTensor2MPI_NCCLKernels extends DistributeTensor2MPI_NCCLBase with CudnnUtils with CudaLibs {
```

Comment:
do we want to rename this trait (and the filename) to `DistributeTensor2MPI_NCCLMiscs`? 
it is good to keep the names sync.

Reply:
fixed